examples/errors.js

```diff
@@ -7,19 +7,16 @@ const { emitWarning } = require('process')
 // Emit an `uncaughtException` event
 const uncaughtException = function() {
   setTimeout(() => {
-    // eslint-disable-next-line max-nested-callbacks
-    setTimeout(() => {
-      throw new Error('File not found')
-    }, 0)
+    throw new Error('File not found')
   }, 0)
 }
 
 // Emit `unhandledRejection` and `rejectionHandled` events
 const unhandledRejection = function() {
   const promise = Promise.reject(new Error('Invalid permission'))
   setTimeout(() => {
-    // eslint-disable-next-line no-empty-function, max-nested-callbacks
-    promise.catch(() => {})
+    // eslint-disable-next-line max-nested-callbacks
+    promise.catch(() => ({}))
```

Comment:
There is no need to return an empty object since this promise is not returned. It's only used for the side-effect.

It's ok if the last two functions create ESLint warnings. This file is meant to create side-effects. We can't escape that because of the way process errors work. The ESLint tries to promote functional programming, so it warns on side-effects, but it's ok for those use cases where they are required.